pkg/3scale/amp/operator/apicast_reconciler.go

```diff
@@ -545,6 +555,69 @@ func apicastPodTemplateEnvConfigMapAnnotationsMutator(desired, existing *appsv1.
 	return updated, nil
 }
 
+func (r *ApicastReconciler) reconcileAPImanagerCR(ctx context.Context) (ctrl.Result, error) {
+
+	changed := false
+	changed, err := r.reconcileApimanagerSecretLabels(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if changed {
+		err = r.Client().Update(ctx, r.apiManager)
```

Comment:
IMO we can hit an err here on updating the resource that can be out of sync. I'd like an error handling here instead of just logging.

Reply:
done. Thank you